packages/fxa-content-server/app/scripts/views/base.js

```diff
@@ -381,6 +381,9 @@ var BaseView = Backbone.View.extend({
         err => {
           if (AuthErrors.is(err, 'INVALID_TOKEN')) {
             this.logError(AuthErrors.toError('SESSION_EXPIRED'));
+            // The redirectTo in .navigate() is lost if there's later navigations, so by saving it here
+            // we can get it later (e.g., in case of a signUp):
+            this.relier.set('redirectTo', window.location.href);
```

Comment:
`window.location.href` doesn't seem quite right here, but I'm not sure. `this.currentPage` is another candidate (they'd almost be equivalent, except `this.currentPage` doesn't start with a `/` so it's not really a proper relative URL, but obviously that could be adjusted).

Reply:
`this.currentPage` is probably a good option, if only because [it looks like it does a bit of normalization and strips off leading & trailing slashes](https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/scripts/lib/router.js#L442)?

Reply:
`this.currentPage` only has the pathname, it doesn't contain any of the origin. Is the origin necessary here?

Reply:
> It also seems like we'd need to propagate the current `flowId` and `flowBeginTime` so the flow metrics behave as expected. @philbooth should be able to provide guidance there.

They're both already included in the resume token, is that being used for the redirect you're talking about here? Should just work if so.